doc/source/whatsnew/v0.23.0.txt

```diff
@@ -624,6 +624,7 @@ Removal of prior version deprecations/changes
 - The modules `pandas.tools.hashing` and `pandas.util.hashing` have been removed (:issue:`16223`)
 - The top-level functions ``pd.rolling_*``, ``pd.expanding_*`` and ``pd.ewm*`` have been removed (Deprecated since v0.18).
   Instead, use the DataFrame/Series methods :attr:`~DataFrame.rolling`, :attr:`~DataFrame.expanding` and :attr:`~DataFrame.ewm` (:issue:`18723`)
+- Imports from ``pandas.core.common`` for api like functions such as ``is_datetime64_dtype`` are now removed. These are located in ``pandas.api.types``. Furthermore, errors are removed as well from ``pandas.core.common`` and are available in ``pandas.errors`` (:issue:`13990`)
```

Comment:
Can remove "for api" and just have "functions like is_datetime64_dtype."

And using the errors here didn't actually emit deprecation warnings did it? Those should be deprecated before removing.

Reply:
no, we simply have to remove them. these are classes and we were never able to actually deprecate them.

Reply:
You can deprecate pretty easily with subclassing

```
class PerformanceWarning(pandas.errors.PerformanceWarning):
    def __init__(self, *args, **kwargs):
        warnings.warn('Deprecated. Use pandas.errors instead', FutureWarning)
        super().__init__(*args, **kwargs)
```

could meta-program that if you want, but it's just 4 errors.

Reply:
I believe @jorisvandenbossche tried this, and we had a discussion about this in the original issue.

Reply:
The problem we discussed AFAIR was about the problem people importing an error to then only catch it. That way the error is never 'called', and a deprecation warning like the above would not work. 
It does work for raising such a warning (but I think catching will be more common).

Reply:
I think the only way this can really be solved is by renaming `core/common.py` to eg `core/utils.py`, and then you can deprecate the whole `core/common.py` module, and already raise warning on importing it.

Reply:
Could override isinstancecheck? Probably overkill.

Reply:
yeah I think we had this discussion a while back, and it looks like it can be done, but then the details kill us. @jorisvandenbossche soln would work, but I think will just put a big warning.